src/pages/Pool/index.tsx

```diff
@@ -36,45 +37,61 @@ const mobileTabs = [
 ];
 const LIMIT = 8;
 
-export interface PoolExtended extends Pool {
-  pair: PairExtended;
-  hasBalance: boolean;
-}
-
 function PoolPage() {
-  const location = useLocation();
   const screenClass = useScreenClass();
   const [selectedTimeBase, setSelectedTimeBase] = useState(timeBaseOptions[0]);
   const [currentPage, setCurrentPage] = useState(1);
-  const { pairs, findPair } = usePairs();
-  const api = useAPI();
+  const { findPair, getPair } = usePairs();
   const { bookmarks } = usePairBookmark();
-  const [pools, setPools] = useState<PoolExtended[]>([]);
+  const { pools } = usePools();
   const [selectedTabIndex, setSelectedTabIndex] = useState(0);
 
   const [addresses, setAddresses] =
     useState<[string | undefined, string | undefined]>();
   const [selectedPair, setSelectedPair] = useState<PairExtended>();
+  const api = useAPI();
+
+  const balanceQueryResults = useQueries({
+    queries:
+      pools?.map((pool) => ({
+        queryKey: ["pool", pool.address],
+        queryFn: async () => {
+          const lpAddress = getPair(pool.address)?.liquidity_token;
+          if (lpAddress) {
+            const balance = await api.getTokenBalance(lpAddress);
+            return balance;
+          }
+          return "0";
+        },
+        enabled: !!pool.address,
+        refetchInterval: 30000,
```

Comment:
I guess 3000? seems better to use constant

Reply:
I thought querying for all balances every 3 seconds would be too many requests.
how about 10 seconds then?

Reply:
sorry, it's pools' balances! it's ok for 30 sec IMO